### PR TITLE
Add .editorconfig [skip ci]

### PR DESCRIPTION
**Pull request by pana1990**
EditorConfig helps developers define and maintain consistent coding styles between different editors and IDEs.

For more details [editorconfig](http://editorconfig.org/)

### CODE REVIEW

**Comment by samdark** [comment]
Haven't heard that any major IDEs are using it.

**Comment by pana1990** [comment]
phpstorm, sublime Text, netbeans,... you see this link for more details http://editorconfig.org/#download



[symfony](https://github.com/symfony/symfony/blob/2.8/.editorconfig) and [laravel](https://github.com/laravel/framework/blob/5.1/.editorconfig) are using editorconfig too

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (samdark) [inline .editorconfig]: Shouldn't it be true?

Comment (pana1990) [inline .editorconfig]: No, you see the following scenario : 
```
hello
world
```
Output: 
-> hello world

```
hello__
world
```
Output: 
-> hello 
world
 (where underscores are pretending to be spaces)

For more details this [here](http://area51.phpbb.com/phpBB/viewtopic.php?t=46077)

**Comment by samdark** [comment]
Merged. I wasn't aware of it. Thanks!

**Comment by pana1990** [comment]
@samdark read last post please

**Comment by samdark** [comment]
Oh...

File: .editorconfig (hunk 1 above)
Comment (samdark) [inline .editorconfig]: https://github.com/yiisoft/yii2/commit/daa632495c22c445e09e0b964dcb02eac2fc1adb

**Comment by samdark** [comment]
Fixed.

**Comment by pana1990** [comment]
thank you very much :)

**Comment by pana1990** [comment]
@samdark yii-extensions should have this file too, imo.

i will send pr?

**Comment by samdark** [comment]
OK/

**Pull request closed without merging**